langchain/src/chains/openai_functions/openapi.ts

```diff
@@ -0,0 +1,425 @@
+import { ChatCompletionFunctions } from "openai";
+import { JsonSchema7ObjectType } from "zod-to-json-schema/src/parsers/object.js";
+import { OpenAPIV3_1 } from "openapi-types";
+import Document = OpenAPIV3_1.Document;
+import ParameterObject = OpenAPIV3_1.ParameterObject;
+import SchemaObject = OpenAPIV3_1.SchemaObject;
+
+import { OpenAPISpec } from "../../util/openapi.js";
+import { ChainValues } from "../../schema/index.js";
+import { CallbackManagerForChainRun } from "../../callbacks/manager.js";
+import { BaseChain } from "../base.js";
+import { LLMChain, LLMChainInput } from "../llm_chain.js";
+import { ChatOpenAI } from "../../chat_models/openai.js";
+import { BasePromptTemplate } from "../../prompts/base.js";
+import {
+  ChatPromptTemplate,
+  HumanMessagePromptTemplate,
+} from "../../prompts/chat.js";
+import { SequentialChain } from "../sequential_chain.js";
+import { JsonOutputFunctionsParser } from "../../output_parsers/openai_functions.js";
+
+type OpenAPIExecutionMethod = (
+  name: string,
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  requestArgs: Record<string, any>,
+  options?: {
+    headers?: Record<string, string>;
+    params?: Record<string, string>;
+  }
+) => Promise<Response>;
+
+function formatURL(url: string, pathParams: Record<string, string>): string {
+  const expectedPathParamNames = [...url.matchAll(/{(.*?)}/g)].map((match) => match[1]);
```

Comment:
`matchAll` is ok right @nfcampos?

Reply:
@jacoblee93 I have been using: `createOpenAPIChain` to call my open-api given spec I have.
When Im using: `verbose: true`, I see all the chain phases.
Write before calling my API, I see that The chain takes this function argument:
```
[chain/start] [1:chain:SequentialChain > 4:chain:SimpleRequestChain] Entering Chain run with input: {
  "query": "Show me all my organization hidden Users",
  "function": {
    "name": "getUsersByQueryParams",
    "arguments": {
      "params": {
        "hidden": {
          "value": true
        }
      }
    }
  }
}
```

and Unfortunately runs the query with: `{ hidden: { value: true } }`, instead of just: `{ hidden: true }`.

Any suggestion on how to take care of this Issue? thanks 🙏🏼